app/services/katello/host/package_profile_uploader.rb

```diff
@@ -0,0 +1,41 @@
+module Katello
+  module Host
+    class PackageProfileUploader
+      def initialize(profile_string:, host:)
+        @profile_string = profile_string
+        @host = host
+      end
+
+      def upload
+        profile = JSON.parse(@profile_string)
+        #free the huge string from the memory
+        @profile_string = 'TRIMMED'.freeze
+        import_package_profile(profile)
+      end
+
+      def import_package_profile(profile)
+        self.class.import_package_profile_for_host(@host.id, profile)
+      end
+
+      def trigger_applicability_generation
```

Comment:
I kept `trigger_applicability_generation` as two un-DRY methods in each class. My reasoning process was

1. I should probably DRY this up and have both classes call a single `trigger_applicability_generation`
2. Which class should it live in? Probably this one, since it already has another shared method (`import_package_profile_for_host`)
3. But we're going to delete this class eventually anyway, so maybe it should go in the other one?
4. But if it goes in `ProfilesUploader`, I'd have one shared method there and one shared method here. Makes more sense to pick one of the classes to have all the shared methods
5. It would make more sense for the shared methods to live in the class that's not getting deleted eventually. So it should live in `ProfilesUploader` and not here.
6. But it's a lot of work to move the logic for the other shared method over there (import_package_profile_for_host)
7. Also, `@host` might be nil in that class, but will never be nil here.
8. So it's just easier to have two `trigger_applicability_generation` methods, one for each class, and leave the rest as is.